pcsx2/IPC.cpp

```diff
@@ -0,0 +1,304 @@
+	if (m_sock < 0)
+	{
+		Console.WriteLn(Color_Red, "IPC: Cannot open socket! Shutting down...");
+		return;
+	}
+	server.sun_family = AF_UNIX;
+	strcpy(server.sun_path, SOCKET_NAME);
+
+	// we unlink the socket so that when releasing this thread the socket gets
+	// freed even if we didn't close correctly the loop
+	unlink(SOCKET_NAME);
+	if (bind(m_sock, (struct sockaddr*)&server, sizeof(struct sockaddr_un)))
+	{
+		Console.WriteLn(Color_Red, "IPC: Error while binding to socket! Shutting down...");
+		return;
+	}
+#endif
+
+	// maximum queue of 100 commands before refusing
+	listen(m_sock, 100);
+
+	// we save a handle of the main vm object
+	m_vm = vm;
+
+	// we start the thread
+	Start();
+}
+
+void SocketIPC::ExecuteTaskInThread()
+{
+	char buf[1024];
+	int msgsock = 0;
+	while (true)
+	{
+		msgsock = accept(m_sock, 0, 0);
+		if (msgsock == -1)
+		{
+			return;
+		}
+		else
+		{
+			bzero(buf, sizeof(buf));
+#ifdef _WIN32
+			if (recv(msgsock, buf, 1024, 0) < 0)
+			{
+#else
+			if (read(msgsock, buf, 1024) < 0)
+			{
+#endif
+				return;
+			}
+			else
+			{
+				auto res = ParseCommand(buf);
+#ifdef _WIN32
+				if (send(msgsock, res.second, res.first, 0) < 0)
+				{
+#else
+				if (write(msgsock, res.second, res.first) < 0)
+				{
+#endif
+					return;
+				}
+			}
+		}
+	}
+}
+
+SocketIPC::~SocketIPC()
+{
+#ifdef _WIN32
+	closesocket(m_sock);
+	WSACleanup();
+#else
+	close(m_sock);
+	unlink(SOCKET_NAME);
+#endif
+	// destroy the thread
+	try
+	{
+		pxThread::Cancel();
+	}
+	DESTRUCTOR_CATCHALL
+}
+
+char* SocketIPC::MakeOkIPC(int size)
+{
+	char* res_array = (char*)malloc(size * sizeof(char));
+	res_array[0] = (unsigned char)IPC_OK;
+	return res_array;
+}
+
+char* SocketIPC::MakeFailIPC(int size)
+{
+	char* res_array = (char*)malloc(size * sizeof(char));
+	res_array[0] = (unsigned char)IPC_FAIL;
+	return res_array;
+}
+
+std::pair<int, char*> SocketIPC::ParseCommand(char* buf)
+{
+	//         IPC Message event (1 byte)
+	//         |  Memory address (4 byte)
+	//         |  |           argument (VLE)
+	//         |  |           |
+	// format: XX YY YY YY YY ZZ ZZ ZZ ZZ
+	//        reply code: 00 = OK, FF = NOT OK
+	//        |  return value (VLE)
+	//        |  |
+	// reply: XX ZZ ZZ ZZ ZZ
+	IPCCommand opcode = (IPCCommand)buf[0];
+	// YY YY YY YY from schema above
+	u32 a = FromArray<u32>(buf, 1);
+	std::pair<int, char*> rval;
+	switch (opcode)
+	{
+		case MsgRead8:
+		{
+			u8 res;
+			if (m_vm->HasActiveMachine() == true)
+			{
+				res = memRead8(a);
+			}
+			else
+			{
+				goto error;
+			}
+			rval = std::make_pair(2, ToArray(MakeOkIPC(2), res, 1));
+			break;
+		}
+		case MsgRead16:
+		{
+			u16 res;
+			if (m_vm->HasActiveMachine() == true)
+			{
+				res = memRead16(a);
+			}
+			else
+			{
+				goto error;
+			}
+			rval = std::make_pair(3, ToArray(MakeOkIPC(3), res, 1));
+			break;
+		}
+		case MsgRead32:
+		{
+			u32 res;
+			if (m_vm->HasActiveMachine() == true)
+			{
+				res = memRead32(a);
+			}
```

Comment:
Don't really need brackets for 1 code lines.

Reply:
Just fixed it for IPC{.cpp,.h}

Reply:
I meant remove the brackets altogether, otherwise it was fine how clang formatted.
```
			if (m_vm->HasActiveMachine() == true)
				res = memRead32(a);
```

Reply:
Ah! I don't usually use bracketless if because of bugs they can cause, let me fix that